store/localcachelayer/channel_layer.go

```diff
@@ -132,3 +152,43 @@ func (s LocalCacheChannelStore) GetPinnedPostCount(channelId string, allowFromCa
 
 	return count, nil
 }
+
+func (s LocalCacheChannelStore) SaveMember(member *model.ChannelMember) (*model.ChannelMember, *model.AppError) {
+	defer s.InvalidateAllChannelMembersForUser(member.UserId)
+	return s.ChannelStore.SaveMember(member)
+}
+
+func (s LocalCacheChannelStore) IsUserInChannelUseCache(userId string, channelId string) bool {
+	ids, err := s.GetAllChannelMembersForUser(userId, true, false)
+	if err != nil {
+		mlog.Error("Error getting all channel members for user", mlog.Err(err))
+		return false
+	}
+
+	if _, ok := ids[channelId]; ok {
+		return true
+	}
+
+	return false
+}
+
+func (s LocalCacheChannelStore) GetAllChannelMembersForUser(userId string, allowFromCache bool, includeDeleted bool) (map[string]string, *model.AppError) {
+	cache_key := userId
```

Comment:
I think can be more clear to start with:

```go
if !allowFromCache {
    return s.ChannelStore.GetAllChannelMembersForUser(userId, allowFromCache, includeDeleted)
}
```

And them assume that you are using allow from cache in the rest of the code